include/pm/device.h

```diff
@@ -77,6 +80,12 @@ enum pm_device_action {
 	PM_DEVICE_ACTION_TURN_OFF,
 	/** Force suspend. */
 	PM_DEVICE_ACTION_FORCE_SUSPEND,
+	/** Handle power domain resume */
+	PM_DEVICE_ACTION_HANDLE_PD_RESUME,
```

Comment:
I don't believe that it is necessary to create 3 new actions to support all the transitions that are possible.
Ignoring the `PM_DEVICE_STATE_SUSPENDING` state, there are 3 stable states for devices to exist in.

1. `PM_DEVICE_STATE_ACTIVE`
2. `PM_DEVICE_STATE_SUSPENDED`
3. `PM_DEVICE_STATE_OFF`

The actions `PM_DEVICE_ACTION_SUSPEND` and `PM_DEVICE_ACTION_RESUME` handle transitions between 1 and 2.
`PM_DEVICE_ACTION_TURN_OFF` handles 2 -> 3, so IMO we only need a `PM_DEVICE_ACTION_TURN_ON` to handle 3 -> 2.

Reply:
That makes sense, the two proposed actions were not related to devices but just a way to notify about power domain transitions. I think just adding `TURN_ON` action is a better solution indeed.